### PR TITLE
EVG-6945 generate device name better

### PR DESCRIPTION
**Pull request by ablack12**
Generates device name differently because the AWS docs seem inconsistent, and we should regenerate device name on failure so we don't retry repeatedly with a bad device name. Also some code cleanup (using the connector for volume things, remove old debugging, etc.)

### CODE REVIEW

File: cloud/ec2_util.go [hunk 1]
```diff
@@ -299,12 +299,35 @@ type Terms struct {
...
+			deviceName = generateDeviceNameForVolume(isWindowsHost)
+			exists, err := host.HostExistsWithVolumeWithDeviceName(deviceName)
+			if err != nil {
+				return true, errors.Wrapf(err, "error checking if device name already exists")
+			}
+			if !exists {
+				return false, nil
+			}
+			return true, errors.New("generated device name already exists")
+		}, 500, 0, 0)
```
Comment (tychoish) [inline cloud/ec2_util.go]: I'm not sure what passing 0 for min/max duration are meant to do. Looking at our code, we start at 100ms sleeps (overriding shorter set values for min) and when max is 0 we do (min)*2^<attempts>, which is 500, so five seconds? is that what we want?

File: cloud/ec2_util.go [hunk 2]
```diff
@@ -299,12 +299,35 @@ type Terms struct {
...
+
+	return deviceName, err
+}
+
+// formats /dev/sd[f-p]and xvd[f-p] taken from https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html
+func generateDeviceNameForVolume(isWindowsHost bool) string {
 	letters := "fghijklmnop"
 	rand.Seed(time.Now().Unix())
```
Comment (tychoish) [inline cloud/ec2_util.go]: we can probably drop this line?

Comment (ablack12) [reply cloud/ec2_util.go]: Why? We do still use rand.

File: cloud/ec2_util.go (hunk 1 above)
Comment (ablack12) [reply cloud/ec2_util.go]: Time between retries for this wouldn't matter, so no. I'll change it.

**Pull request merged**